### ISSUE TITLE
C++17 auto in function template triggers false-positive error

### ISSUE THREAD

**Issue by taylorgordon20** [microsoft/vscode-cpptools#2478]
```
struct Foo {
  template <auto x>
  void foo() {}
};

void bar() {
}

Foo f;
f.foo<bar>();
```

Results in "f" being underlined with an error squiggle and the message:

`no instance of function template "Foo::foo" matches the argument list -- object type is: Foo`

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. It repros with VS 2017, so I've reported it there: https://developercommunity.visualstudio.com/content/problem/327851/c17-auto-in-function-template-triggers-false-posit.html .

UPDATE: The fix will be in our next Insiders release (0.21.0-insiders), probably next week.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.21.0-insiders (auto installed if `"C_Cpp.updateChannel": "Insiders"` is set, or the [offline vsix](https://github.com/Microsoft/vscode-cpptools/releases/tag/0.21.0-insiders)). Let us know if you find any problems with it.
